connector/connect/server/src/main/scala/org/apache/spark/sql/connect/planner/StreamingForeachBatchHelper.scala

```diff
@@ -31,6 +38,17 @@ object StreamingForeachBatchHelper extends Logging {
 
   type ForeachBatchFnType = (DataFrame, Long) => Unit
 
+  case class RunnerCleaner(runner: StreamingPythonRunner) extends AutoCloseable {
+    override def close(): Unit = {
+      try runner.stop()
+      catch {
+        case NonFatal(ex) =>
+          logWarning("Error while stopping streaming Python worker", ex)
+        // Exception is not propagated.
```

Comment:
The indentation is due to scalafmt. I will move it up so that it is aligned better. 

`NonFatal` is a good practice to catch only application error rather than some JVM related fatal errors. This is important especially while swallowing the exceptions. See Spark codestyle guide: https://github.com/databricks/scala-style-guide#exception-handling-try-vs-try